### PR TITLE
Generate path for FlexEntity and class_alias

### PR DESCRIPTION
**Pull request by thehawk970**

### CODE REVIEW

File: Bin/Stub.php [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        $verbose    = false;
+
+        while(false !== $c = $this->getOption($v)) switch($c) {
+
+            case 'V':
+                $verbose = true;
+                break;
+
+            case 'd':
+                $porcelaine = true;
```
Comment (Hywan) [inline Bin/Stub.php]: `porcelain`, not `porcelaine`, and it's not that, it's `dryRun`. `porcelain` is different :-).

File: Bin/Stub.php [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Devtools\Bin;
+
+use Hoa\Console;
```
Comment (Hywan) [inline Bin/Stub.php:39]: `use Hoa\File;`

File: Bin/Stub.php [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                $this->resolveOptionAmbiguity($v);
+              break;
+        }
+
+        $hoaPath = 'hoa://Library/';
+
+        if($stub === null)
+            return $this->usage();
+
+        $finder = new \Hoa\File\Finder();
```
Comment (Hywan) [inline Bin/Stub.php]: `$finder = new File\Finder();`

File: Bin/Stub.php [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        $finder
+            ->in($hoaPath)
+            ->name('#\.php$#')
+            ->maxDepth(100)
+            ->files();
+
+        $f = array();
```
Comment (Hywan) [inline Bin/Stub.php]: Use short-array syntax here :-).

File: Bin/Stub.php [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+        // READ
+        foreach ($finder as $key => $value) {
```
Comment (Hywan) [inline Bin/Stub.php]: We never use `$key`, do not declare it :-).

File: Bin/Stub.php (hunk 2 above)
Comment (Hywan) [inline Bin/Stub.php:39]: `use Hoa\Core;`

File: Bin/Stub.php [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+                if (isset($classAlias[1]) and $classAlias[1] !== '') {
+                    $name  = substr(trim($classAlias[1]), 1, -1);
+                    $end   = substr(trim($classAlias[2]), 1, -1);
+                }
+                 
+                preg_match('#flexEntity\(\'(.*)\'#', $raw, $flexEntity);
+
+                if (count($flexEntity) > 0) {
+                    $c   = $flexEntity[1];
+                    $end = \Hoa\Core\Consistency\Consistency::getEntityShortestName($c);
```
Comment (Hywan) [inline Bin/Stub.php]: `$end = Core\Consistency…;`

File: Bin/Stub.php [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+                }
+
+                if ($end !== '') {
+                    $f[] = array('class' => str_replace('/', '\\', $name), 'alias' => $end);
```
Comment (Hywan) [inline Bin/Stub.php]: Use array short syntax :-).

File: Bin/Stub.php [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+ * Class \Hoa\Devtools\Bin\Stub
+ *
+ * This command resolve the class_alias function in destination to IDE.
+ *
+ * @author     Julien CLAUZEL <julien.clauzel@hoa-project.net>
+ * @copyright  Copyright © 2007-2015 Ivan Enderlin.
+ * @license    New BSD License
+ */
+
+class Stub extends \Hoa\Console\Dispatcher\Kit {
```
Comment (Hywan) [inline Bin/Stub.php]: The name says nothing about what it really does :-/. (do not forget to update the help too).
Suggestion (but I am not sure!): `UnfoldClassAliases`?

**Comment by Hywan** [comment]
@camael24 Could you provide a description and an example of use please? Also, I am not certain to be the one to evaluate this PR. Do you know someone else who could do a peer-review?

**Comment by thehawk970** [comment]
@Hywan thanks to the review, i will apply those modification (not sure "ces modifications") tomorrow, today I too budy

The PR its to help the class_alias resolution in destination to our IDE, we juste need to include the generated file in exemple `/tmp/foo.php` in the IDE.

An output exemple file is : 

```
<?php 
namespace Hoa\Console {
class Readline extends \Hoa\Console\Readline\Readline {}
}
namespace Hoa\Console\Readline {
class Autocompleter extends \Hoa\Console\Readline\Autocompleter\Autocompleter {}
}
namespace Hoa {
class Console extends \Hoa\Console\Console {}
}
namespace Hoa {
class Prototype extends \Hoa\Prototype\Prototype {}
}
namespace Hoa {
class Memory extends \Hoa\Memory\Memory {}
}
namespace Hoa\Fastcgi {
class Exception extends \Hoa\Fastcgi\Exception\Exception {}
}
namespace Hoa\Cache {
class Backend extends \Hoa\Cache\Backend\Backend {}
}
namespace Hoa {
class Cache extends \Hoa\Cache\Cache {}
}
namespace Hoa\Cache {
class Frontend extends \Hoa\Cache\Frontend\Frontend {}
}
namespace Hoa\Core {
class Parameter extends \Hoa\Core\Parameter\Parameter {}
}
namespace Hoa\Core {
class Event extends \Hoa\Core\Event\Event {}
}
namespace Hoa\Core {
class Exception extends \Hoa\Core\Exception\Exception {}
}
namespace Hoa\Core {
class Consistency extends \Hoa\Core\Consistency\Consistency {}
}
namespace Hoa\Core {
class Protocol extends \Hoa\Core\Protocol\Protocol {}
}
namespace Hoa {
class Core extends \Hoa\Core\Core {}
}
namespace Hoa\Core {
class Data extends \Hoa\Core\Data\Data {}
}
namespace Hoa\Http {
class Response extends \Hoa\Http\Response\Response {}
}
namespace Hoa\Http {
class Exception extends \Hoa\Http\Exception\Exception {}
}
namespace Hoa {
class Http extends \Hoa\Http\Http {}
}
namespace Hoa\Database\Layer {
class Pdo extends \Hoa\Database\Layer\Pdo\Pdo {}
}
namespace Hoa\Database {
class Query extends \Hoa\Database\Query\Query {}
}
namespace Hoa {
class Praspel extends \Hoa\Praspel\Praspel {}
}
namespace Hoa\Praspel\Exception {
class Failure extends \Hoa\Praspel\Exception\Failure\Failure {}
}
namespace Hoa\Praspel {
class Exception extends \Hoa\Praspel\Exception\Exception {}
}
namespace Hoa\Praspel\Iterator {
class Coverage extends \Hoa\Praspel\Iterator\Coverage\Coverage {}
}
namespace Hoa\Praspel\Model {
class Variable extends \Hoa\Praspel\Model\Variable\Variable {}
}
namespace Hoa\Praspel {
class AssertionChecker extends \Hoa\Praspel\AssertionChecker\AssertionChecker {}
}
namespace Hoa {
class Session extends \Hoa\Session\Session {}
}
namespace Hoa\Session {
class Exception extends \Hoa\Session\Exception\Exception {}
}
namespace Hoa {
class Graph extends \Hoa\Graph\Graph {}
}
namespace Hoa {
class Locale extends \Hoa\Locale\Locale {}
}
namespace Hoa\Locale {
class Localizer extends \Hoa\Locale\Localizer\Localizer {}
}
namespace Hoa {
class Tree extends \Hoa\Tree\Tree {}
}
namespace Hoa\Math\Combinatorics {
class Combination extends \Hoa\Math\Combinatorics\Combination\Combination {}
}
namespace Hoa\Math {
class Exception extends \Hoa\Math\Exception\Exception {}
}
namespace Hoa\Math {
class Sampler extends \Hoa\Math\Sampler\Sampler {}
}
namespace Hoa\Compiler {
class Exception extends \Hoa\Compiler\Exception\Exception {}
}
namespace Hoa\Compiler\Llk {
class Rule extends \Hoa\Compiler\Llk\Rule\Rule {}
}
namespace Hoa\Compiler\Llk {
class Sampler extends \Hoa\Compiler\Llk\Sampler\Sampler {}
}
namespace Hoa\Compiler {
class Llk extends \Hoa\Compiler\Llk\Llk {}
}
namespace Hoa {
class Ruler extends \Hoa\Ruler\Ruler {}
}
namespace Hoa\Ruler {
class Exception extends \Hoa\Ruler\Exception\Exception {}
}
namespace Hoa\Ruler\Model {
class Bag extends \Hoa\Ruler\Model\Bag\Bag {}
}
namespace Hoa\Ruler {
class Model extends \Hoa\Ruler\Model\Model {}
}
namespace Hoa\Xyl {
class Element extends \Hoa\Xyl\Element\Element {}
}
namespace Hoa\Xyl\Interpreter {
class Html extends \Hoa\Xyl\Interpreter\Html\Html {}
}
namespace Hoa\Xyl {
class Interpreter extends \Hoa\Xyl\Interpreter\Interpreter {}
}
namespace Hoa {
class Xyl extends \Hoa\Xyl\Xyl {}
}
namespace Hoa\Realdom {
class Exception extends \Hoa\Realdom\Exception\Exception {}
}
namespace Hoa {
class Realdom extends \Hoa\Realdom\Realdom {}
}
namespace Hoa {
class Promise extends \Hoa\Promise\Promise {}
}
namespace Hoa {
class Bench extends \Hoa\Bench\Bench {}
}
namespace Hoa {
class Log extends \Hoa\Log\Log {}
}
namespace Hoa\Log {
class Backtrace extends \Hoa\Log\Backtrace\Backtrace {}
}
namespace Hoa {
class String extends \Hoa\String\String {}
}
namespace Hoa\Xml {
class Element extends \Hoa\Xml\Element\Element {}
}
namespace Hoa\Xml {
class Exception extends \Hoa\Xml\Exception\Exception {}
}
namespace Hoa {
class Xml extends \Hoa\Xml\Xml {}
}
namespace Hoa {
class Iterator extends \Hoa\Iterator\Iterator {}
}
namespace Hoa\Iterator {
class Recursive extends \Hoa\Iterator\Recursive\Recursive {}
}
namespace Hoa {
class   extends \Hoa\ \  {}
}
namespace Hoa {
class Dispatcher extends \Hoa\Dispatcher\Dispatcher {}
}
namespace Hoa\Mail {
class Exception extends \Hoa\Mail\Exception\Exception {}
}
namespace Hoa\Mail {
class Content extends \Hoa\Mail\Content\Content {}
}
namespace Hoa {
class Registry extends \Hoa\Registry\Registry {}
}
namespace Hoa {
class Model extends \Hoa\Model\Model {}
}
namespace Hoa\Router {
class Exception extends \Hoa\Router\Exception\Exception {}
}
namespace Hoa {
class Router extends \Hoa\Router\Router {}
}
namespace Hoa\Stream {
class Filter extends \Hoa\Stream\Filter\Filter {}
}
namespace Hoa {
class Stream extends \Hoa\Stream\Stream {}
}
namespace Hoa\Stream {
class Wrapper extends \Hoa\Stream\Wrapper\Wrapper {}
}
namespace Hoa\Stream\Wrapper {
class IWrapper extends \Hoa\Stream\Wrapper\IWrapper\IWrapper {}
}
namespace Hoa\Socket {
class Connection extends \Hoa\Socket\Connection\Connection {}
}
namespace Hoa {
class Socket extends \Hoa\Socket\Socket {}
}
namespace Hoa {
class Translate extends \Hoa\Translate\Translate {}
}
namespace Hoa {
class File extends \Hoa\File\File {}
}
namespace Hoa\File {
class Exception extends \Hoa\File\Exception\Exception {}
}
namespace Hoa\File {
class Link extends \Hoa\File\Link\Link {}
}
namespace Hoa\File {
class Temporary extends \Hoa\File\Temporary\Temporary {}
}
namespace Hoa {
class Acl extends \Hoa\Acl\Acl {}
}
namespace Hoa\Websocket {
class Exception extends \Hoa\Websocket\Exception\Exception {}
}
namespace Hoa\XmlRpc {
class Exception extends \Hoa\Xmlrpc\Exception\Exception {}
}
namespace Hoa\XmlRpc {
class Message extends \Hoa\Xmlrpc\Message\Message {}
}
namespace Hoa {
class Stringbuffer extends \Hoa\Stringbuffer\Stringbuffer {}
}
namespace Hoa {
class Mime extends \Hoa\Mime\Mime {}
}
namespace Hoa\Mime {
class Exception extends \Hoa\Mime\Exception\Exception {}
}
namespace Hoa {
class Zombie extends \Hoa\Zombie\Zombie {}
}
```

File: Bin/Stub.php (hunk 8 above)
Comment (thehawk970) [inline Bin/Stub.php]: I apply the same logic than atoum with the atoum-stub-generator, but i can do an error  :)

**Comment by Hywan** [comment]
@camael24 “these modifications” :-).

File: Bin/Stub.php (hunk 8 above)
Comment (Hywan) [inline Bin/Stub.php]: But we are not atoum :-).

**Comment by Hywan** [comment]
@camael24 I don't use an IDE but does it fit in the IDE's users' workflow?

**Comment by Metalaka** [comment]
I think it can be useful (because I'm very lazy so I need a good auto-completion :).
But why adding support for IDE now when you plan to remove PHPDoc in favor of Praspel ?
(related to: https://github.com/hoaproject/Core/issues/65)

What about adding abstract/interface/trait/final ?

```
namespace Hoa {
abstract class Dispatcher extends \Hoa\Dispatcher\Dispatcher {}
}
namespace Hoa {
interface Router extends \Hoa\Router\Router {}
}
```

Could be done with [ReflexionClass::is*()](http://php.net/manual/en/class.reflectionclass.php).

**Comment by Hywan** [comment]
@Metalaka This PR address a different problem than the (hypothetical) removal PHPDoc.

**Comment by Hywan** [comment]
ping?

File: Bin/Stub.php (hunk 7 above)
Comment (Metalaka) [inline Bin/Stub.php]: To generate more specific/valid alias you can add a regex and use the `keyword` value on line 169.

``` php
$c = substr($end, strrpos($end, '\\') + 1);
preg_match('#((?:abstract\s)?class|interface|trait)\s'.$c.'\s#', $raw, $keyword);
$f[] = [
    'class'   => str_replace('/', '\\', $name),
    'alias'   => $end,
    'keyword' => $keyword[1]
    // 'classname' => $c // reused on line 166.
];
```

So the generated code will be valid, until Hoa don't use `final` class ;)

Comment (Metalaka) [inline Bin/Stub.php]: In Hoa, there may be more than one space between `class` keyword and class name.
eg: [`Hoa\Http\Response`](https://github.com/hoaproject/Http/blob/master/Response/Response.php#L82).

``` php
'#((?:abstract\s)?class|interface|trait)\s+'.$c.'\s#'
```

(However it's not [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#41-extends-and-implements).)

File: Bin/Stub.php [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+            case '?':
+                return $this->usage();
+              break;
+
+            case '__ambiguous':
+                $this->resolveOptionAmbiguity($v);
+              break;
+        }
+
+        $hoaPath = 'hoa://Library/';
```
Comment (Metalaka) [inline Bin/Stub.php]: Hoa will resolve this in the following order and use the first it found.
- Data/Library/Hoathis/
- Data/Library/Hoa/
- vendor/Hoathis/
- vendor/Hoa/

So if Hoa find one of the three first directory the script will not work as expected.

Moreover, it cause problems with Windows :/

**Comment by Metalaka** [comment]
Fix https://github.com/hoaproject/Core/issues/39

I worked on this today: https://github.com/hoaproject/Devtools/compare/master...Metalaka:alias
- CS fixed.
- Will generate more specific aliases (interface/abstract...)
- "read" loop has been refactored.

@Hywan Should I do a PR on `camael24:f/stub_generate` or else ?

**Comment by Hywan** [comment]
@Metalaka Yup, make a PR on this PR. Except if @camael24 is too busy to merge your PR.

@camael24: Thoughts?

**Comment by thehawk970** [comment]
Make the PR on my PR i merge it as soon as possible :)

**Comment by thehawk970** [comment]
I juste rebase my PR on master :) and you can go :)

**Comment by Metalaka** [comment]
Done.

**Comment by Hywan** [comment]
And now?

**Comment by Metalaka** [comment]
/ping @camael24 : https://github.com/camael24/Devtools/pull/1 are waiting you ;)
After that you could merge this PR.

**Comment by thehawk970** [comment]
Merged :) thanks @Metalaka

File: Bin/Stub.php [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+
+/**
+ * Class \Hoa\Devtools\Bin\Stub
+ *
+ * This command resolve the class_alias function in destination to IDE.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+class Stub extends Console\Dispatcher\Kit
```
Comment (Hywan) [inline Bin/Stub.php:51]: We should provide a more explicit name here, isn't it :smiley:?

File: Bin/Stub.php [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+        $dryRun  = false;
+        $stub    = null;
+        $verbose = false;
+
+        while (false !== $c = $this->getOption($v)) {
+            switch ($c) {
+                case 'd':
+                    $dryRun = true;
+
+                  break;
```
Comment (Hywan) [inline Bin/Stub.php]: Add 2 spaces before `break;` (everywhere) please.

File: Bin/Stub.php [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+                case 'V':
+                    $verbose = true;
+
+                    break;
+
+                case 'h':
+                case '?':
+                    return $this->usage();
+
+                  break;
```
Comment (Hywan) [inline Bin/Stub.php]: Remove this `break;`, useless regarding the `return` statement.

File: Bin/Stub.php [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+        $hoaPath = 'hoa://Library/';
+        $aliases = [];
+        $finder  = new File\Finder();
+        $finder
+            ->in(resolve($hoaPath, true, true))
+            ->name('#\.php$#')
+            ->maxDepth(100)
+            ->files();
+
+        // READ
```
Comment (Hywan) [inline Bin/Stub.php]: Useless comment.

File: Bin/Stub.php [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+        foreach ($finder as $file) {
+            $pathName  = $file->getPathName();
```
Comment (Hywan) [inline Bin/Stub.php]: One useless space before `=` symbols.

File: Bin/Stub.php [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+
+            if (true === $verbose) {
+                echo
+                    $keyword[1] . ': ' .
+                    $FQCN . ' > ' .
+                    $alias . "\n";
+            }
+        }
+
+        // WRITE
```
Comment (Hywan) [inline Bin/Stub.php]: Useless comment.

File: Bin/Stub.php [hunk 16]
```diff
@@ -0,0 +1,202 @@
...
+        }
+
+        $out = '<?php ' . "\n";
+        foreach ($aliases as $class) {
+            $ns = substr($class['alias'], 0, strrpos($class['alias'], '\\'));
+
+            $out .= 'namespace ' . $ns . ' {' . "\n";
+            $out .= $class['keyword'] . ' ' . $class['className'];
+            $out .= ' extends \\' . $class['FQCN'] . ' {}' . "\n";
+            $out .= '}' . "\n";
```
Comment (Hywan) [inline Bin/Stub.php]: Simply:

``` php
$out .=
    'namespace …' .
    $class['keyword'] …;
```

File: Bin/Stub.php [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    /**
+     * The command usage.
+     *
+     * @return  void
+     */
+    public function usage()
+    {
+        echo 'Usage   : devtools:stub <options>', "\n",
```
Comment (Hywan) [inline Bin/Stub.php]:
``` php
echo
    'Usage …',
    '…';
```

File: Bin/Stub.php [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+class Stub extends Console\Dispatcher\Kit
+{
+    /**
+     * Options description.
+     *
+     * @var array
+     */
+    protected $options = [
+        ['dry-run', Console\GetOption::NO_ARGUMENT      , 'd'],
+        ['stub',    Console\GetOption::REQUIRED_ARGUMENT, 's'],
```
Comment (Hywan) [inline Bin/Stub.php]: `output` would be better than `stub`, thoughts?

Comment (Hywan) [inline Bin/Stub.php]: And since this is the most important option, it must come first in this list.

**Comment by thehawk970** [comment]
Any proposal for an new name ?

**Comment by Hywan** [comment]
What is the goal of this command (rhetoric question)?

File: Bin/Stub.php [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+
+
+    /**
+     * The entry method.
+     *
+     * @return  void
+     */
+    public function main()
+    {
+        $dryRun = false;
```
Comment (Metalaka) [inline Bin/Stub.php]: cs: align `=` with `$verbose`

Comment (Hywan) [inline Bin/Stub.php]: `hoa devtools:cs` must fix that normally.

**Comment by thehawk970** [comment]
I run `hoa devtools:cs` :) 

Its for use hoa with an IDE , atoum name that atoum/stub_generator i do the same :p

**Comment by Hywan** [comment]
Well, sorry but the name is not correct. It's not meaningfull. Any proposal?

**Comment by thehawk970** [comment]
What do you think about `autocomplete`, `autocompletion`, 'helper_autocomplete`, 'ide_autocomplete` ? really hard for to to find one word for that :/

**Comment by Hywan** [comment]
What about “unfold classnames”? Or better: “unfold entity names”?

**Comment by thehawk970** [comment]
name are too long for the command no ?
`hoa devtools:unfoldEntityNames`  ?

**Comment by Hywan** [comment]
Nop :-p. Maybe “unfold entites” or “resolve entities” can be correct too.

**Comment by thehawk970** [comment]
I can't see how do this in the code causes how manage the space ? with _ ? there are conflict with \ no ?

File: Bin/Cs.php [hunk 20]
```diff
@@ -124,7 +124,7 @@ public function main()
 
         $arguments[] = $path;
 
-        $processus = new Console\Processus($phpCsFixer, $arguments);
+        $processus = new Console\Processus('php '.$phpCsFixer, $arguments);
```
Comment (Hywan) [inline Bin/Cs.php:127]: Huh?

**Comment by Hywan** [comment]
@camael24 Correct, that's the https://github.com/hoaproject/Core/issues/57 issue. Never mind. Tell me what name you would like to choose and we will see how to format it.

**Comment by Hywan** [comment]
What is the status of this PR?

**Comment by Metalaka** [comment]
@Hywan @camael24 The feature is waiting for a name.

I vote for `resolve entities` or `resolve flex entities`

**Comment by thehawk970** [comment]
i am agree with `resolve flex entities`

**Comment by thehawk970** [comment]
ping @Hywan  see my previous comment

**Comment by Metalaka** [comment]
@camael24 can you go ahead and change the name in the code?
Then @Hywan will merge it (and maybe rename the feature if he has a better name ;)

**Comment by Hywan** [comment]
Resolving implies something that is not solved, so a problem to solve. There is no problem here. What about `unfold flex entities`?

**Comment by Hywan** [comment]
Funny: https://github.com/hoaproject/Devtools/pull/6#issuecomment-123237559.

**Comment by Metalaka** [comment]
@Hywan good for me too

**Comment by Hywan** [comment]
Waiting on @camael24 so :smiley:.

**Comment by Pierozi** [comment]
@camael24 Have you time to finish this PR or did you want some help ?

File: Bin/Stub.php (hunk 10 above)
Comment (Pierozi) [reply Bin/Stub.php:51]: The voted name are `unfoldFlexEntities`

File: Bin/Stub.php [hunk 21]
```diff
@@ -0,0 +1,197 @@
...
+                 'verbose' => 'Echo all information',
+                 'help'    => 'This help.'
+             ]), "\n";
+
+        return;
+    }
+}
+
+__halt_compiler();
+Resolve class_alias function for the IDE.
```
Comment (Pierozi) [inline Bin/Stub.php:197]: A much specific description ? something like
`Create stub class of flex entities for enable auto completion on IDE`

**Review by Pierozi — CHANGES_REQUESTED** [review]
- [ ] Change the class name
- [ ] Increase description

**Comment by jubianchi** [comment]
I don't think the word "unfold" is appropriate here.

Unfolding means, IMHO, that some things are grouped under a common item (folder containing files for example: when we unfold the directory we see files). Here we only have a single item under a single name: flex entity (alias) => actuel class.

I would rather use something like "expand": we go from a short name (flex entity alias) to a full name.

Expanding flex entities is done when we generate stubs of flex entity classes: we could have used "generate stubs" but it wouldn't be explicit about which stubs we generate. "expand flex entities" seems ok to me.

If you need help here, let me know I should have some time in the coming days.

**Comment by Hywan** [comment]
:+1: for expand.

@jubianchi please, feel free to take this PR. I don't have time or energy to take it.

**Comment by Pierozi** [comment]
So let's approve `ExpandFlexEntities` :+1: ( Choose a name can be very difficult )

**Comment by vonglasow** [comment]
👍

**Comment by Pierozi** [comment]
Closed like it was issued on #43

**Pull request closed without merging**